### PR TITLE
Convert civicrm_note.modified_date to timestamp

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
civicrm_note.modified_date is currently a "date" format and does not store the time. This is problematic because sorting in the UI by the default "Modified Date" doesn't necessarily put notes in the right order.

Before
----------------------------------------
`civicrm_note.modified_date` does not store time.

After
----------------------------------------
`civicrm_note.modified_date` stores date and time.

Technical Details
----------------------------------------
As the default for a timestamp field is "now" we don't need to pass in a default to the BAO.

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Upgrade/Incremental/sql/5.23.alpha1.mysql.tpl [hunk 1]
```diff
@@ -9,3 +9,5 @@ ALTER TABLE civicrm_payment_processor ALTER COLUMN is_active SET DEFAULT 1;
 ALTER TABLE civicrm_payment_processor ALTER COLUMN is_test SET DEFAULT 0;
 ALTER TABLE civicrm_payment_processor_type ALTER COLUMN is_active SET DEFAULT 1;
 ALTER TABLE civicrm_payment_processor_type ALTER COLUMN is_default SET DEFAULT 0;
+
+ALTER TABLE civicrm_note MODIFY COLUMN modified_date TIMESTAMP COMMENT 'When was this note last modified/edited';
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/sql/5.23.alpha1.mysql.tpl]: don't we need to set the default as current_timestamp()?

Comment (mattwire) [reply CRM/Upgrade/Incremental/sql/5.23.alpha1.mysql.tpl]: @seamuslee001 MariaDB seems to do that by default .. I've added the default to the PR

**Comment by eileenmcnaughton** [comment]
Test fail relates.

I should note I can't think of a situation where this would cause a problem but I'd like to see a couple more people weigh on on the concept-approved side of this

**Comment by mattwire** [comment]
@colemanw This seems to be failing on the API4 test - any ideas why? With API3 removing the `modified_date` parameter means it uses the database default which is `current_timestamp()` but it looks like api4 is not doing that here?

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by simonjohnparker** [comment]
Hi - this appears to be working correctly on our CiviCRM instance. 

Thanks,

**Comment by irritatie** [comment]
We tested this in the latest CiviCRM release and found no issues. Also did an automated full test of basic CiviCRM functionality and everything worked fine. Thanks!

**Comment by eileenmcnaughton** [comment]
Code & test look OK - merging based on described testing

**Pull request merged**